plugins/ltac/tactic_debug.ml

```diff
@@ -37,8 +37,33 @@ type debug_info =
 let explain_logic_error e = CErrors.print e
 let explain_logic_error_no_anomaly e = CErrors.print_no_report e
 
-let msg_tac_debug s = Proofview.NonLogical.print_debug (s++fnl())
-let msg_tac_notice s = Proofview.NonLogical.print_notice (s++fnl())
+(* Communications with the outside world *)
+module Comm = struct
+
+  (* TODO: ideally we would check that the debugger hooks are
+     correctly set, however we don't do this yet as debugger
+     initialiation is unconditionally done for example in coqc, which
+     doesn't support the debugger interface. Improving this would
+     require some tweaks in tacinterp which are out of scope for the
+     current refactoring. *)
```

Comment:
There are any number of ways to break the code.  Only a small number are both easy to check and worth checking, hence the desire to prove code is correct.  If the hooks are not set, the error is really obvious.  Not sure this is worth trying to check.

Reply:
Note, btw, the it's important to some of my use cases that `coqc` supports `Set Ltac Debug. Set Ltac Batch Debug.`

Reply:
Umm, can you explain a bit more? We could have `coqc` interact with the terminal but it seems weird for a batch compiler to support interactive debugging; I'm not sure I've heard of any other compiler doing so.

Reply:
`Set Ltac Batch Debug.` runs the debugger as if the user entered newline at every prompt, so you get an exhaustive transcript of Ltac tactics. I think this compensates for the weakness of the current debugger.